build.sh

```diff
@@ -335,12 +335,13 @@ download_sync() {
     TMP_DIR="$TMPDIR/sync_bin"
     mkdir -p "${TMP_DIR}"
     SYNC_TMP_TAR="${TMP_DIR}/sync-${REALM_SYNC_VERSION}.tar.xz.tmp"
+    rm -rf "$SYNC_TMP_TAR" # ensure we don't have a corrupt download
```

Comment:
I know this doesn't add caching. It removes the caching feature that's been there for years.

What I was saying is if you want to preemptively delete the archive regardless of whether or not it is valid or corrupt, it breaks caching entirely.

If this is to handle corrupt downloads, it should be done as error handling rather than preemptively.